torchtune/modules/peft/lora.py

```diff
@@ -109,8 +118,38 @@ def adapter_params(self) -> List[str]:
         # NOTE: this function has to be updated if the names of "lora_a" and "lora_b"
         # in this module change.
         adapter_params = ["lora_a.weight", "lora_b.weight"]
+        if self.use_dora:
+            adapter_params.append("lora_m")
         return adapter_params
 
+    def init_dora(self) -> None:
+        # this is a seperate function because,
+        # this should be called after model state dict is called.
+        # But We verify and initialize the model arch first before the loading weights.
+        weight_norm = self._dora_weight_norm
+        self.lora_m.data = weight_norm.data  # Update the data of 'm' directly
+
+    @property
+    def _dora_weight_norm(self) -> Tensor:
+        """
+        Compute the norm of the linear weight and lora adaptor weights.
+        If the base model is quantized, dequantize the weights before computing the norm.
+        Return the norm in NF4 format if the base model is quantized.
+        """
+        weight = self.weight.dequantize() if self._quantize_base else self.weight
+
+        # Perform the operation with regular tensors
+        result = weight + (self.alpha / self.rank) * (
+            self.lora_b.weight @ self.lora_a.weight
+        )
+        norm = torch.linalg.norm(result, dim=1)
+
+        # Clamp the norm to avoid division by zero
+        # TODO(Prakyath): Check with torchtune team whether this should be a parameter ?
```

Comment:
The comment was referring to the clamp line [here](https://github.com/Prakyathkantharaju/torchtune/blob/7b4b8a4dbae45f5b1b436fe902c6ec5006e3e28c/torchtune/modules/peft/lora.py#L147). I have made the minimum of the clamp 1e-6. I was thinking this could be a parameter configured in the yaml file. FYI: I added this clamp to ensure that the denominator of this line is not 0. This is not present in the peft.